src/org/zaproxy/zap/eventBus/SimpleEventBus.java

```diff
@@ -127,31 +128,54 @@ public void registerConsumer(EventConsumer consumer, String publisherName,
 	}
 
 	@Override
-	public void unregisterConsumer(EventConsumer consumer, String publisherName) {
+	public void unregisterConsumer(EventConsumer consumer) {
 		if (consumer == null) {
 			throw new InvalidParameterException("Consumer must not be null");
 		}
 
 		regMgmtLock.lock();
 		try {
-			log.debug("unregisterConsumer " + consumer.getClass().getCanonicalName() + " for " + publisherName);
-			RegisteredPublisher publisher = this.nameToPublisher.get(publisherName);
-			if (publisher != null) {
-				publisher.removeComsumer(consumer);
-			} else {
-				// Check to see if its cached waiting for the publisher
-				for (RegisteredConsumer regCon : this.danglingConsumers) {
-					if (regCon.getConsumer().equals(consumer)) {
-						this.danglingConsumers.remove(regCon);
-						break;
-					}
+			log.debug("unregisterConsumer " + consumer.getClass().getCanonicalName());
```

Comment:
Check `isDebugEnabled()`?

Reply:
It does avoid the creation of the string (less garbage mainly), here does not make much difference since this will not be not called that often I guess.